### PR TITLE
Initial CI Job

### PR DESCRIPTION
**Pull request by moriarty**
Only does a colcon build in the studio container

- rely on github setting workdir to /github/workspace and mounting the code there

### CODE REVIEW

**Comment by moriarty** [comment]
I've gotten to a point where I can't test using https://github.com/nektos/act/ and will peel parts into here

**Comment by moriarty** [comment]
This needs to be squash merged :see_no_evil:

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  integration-test-in-studio-container:
+    runs-on: ubuntu-latest
+    env:
+      USER_WS: /github/workspace
+    # TODO: docker tag
+    # TODO: example for custom base images
+    container: picknikciuser/moveit-studio:main
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run colcon build and test
```
Comment (eholum) [inline .github/workflows/ci.yaml:17]: Maybe worth separating into build and test? It's nice to always get test results, for instance, and this won't do it if the tests fail I think.


```
      - name: Run tests
        run: |
          . /opt/ros/humble/setup.sh
          . /opt/overlay_ws/install/setup.sh
          colcon test
      - if: always()
        run: colcon test-result --verbose
```

Comment (moriarty) [reply .github/workflows/ci.yaml:17]: oh yes actually I have it like that in another branch... I've been peeling off changes to get this in

**Pull request merged**